lib/auth/mocku2f/mocku2f.go

```diff
@@ -84,6 +87,51 @@ type Key struct {
 	counter uint32
 }
 
+var _ json.Marshaler = (*Key)(nil)
+
+func (k *Key) MarshalJSON() ([]byte, error) {
+	type Alias Key
+
+	privateKeyPem, err := keys.MarshalPrivateKey(k.PrivateKey)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	s := struct {
+		PrivateKey string `json:"PrivateKey"`
+		Counter    uint32 `json:"Counter"`
+		*Alias
+	}{
+		PrivateKey: string(privateKeyPem),
+		Counter:    k.counter,
+		Alias:      (*Alias)(k),
+	}
+	return json.Marshal(s)
+}
+
+func (k *Key) UnmarshalJSON(data []byte) error {
+	type Alias Key
+	s := struct {
+		PrivateKey string `json:"PrivateKey"`
+		Counter    uint32 `json:"Counter"`
+		*Alias
+	}{}
+	if err := json.Unmarshal(data, &s); err != nil {
+		return trace.Wrap(err)
+	}
+	privateKey, err := keys.ParsePrivateKey([]byte(s.PrivateKey))
+	if err != nil {
+		return trace.Wrap(err, "parsing private key")
+	}
+	ecPrivateKey, ok := privateKey.Signer.(*ecdsa.PrivateKey)
+	if !ok {
+		return trace.BadParameter("expected ECDSA private key, got %T", privateKey)
+	}
```

Comment:
If we define the field as a []byte containing SEC1 DER we can just use `x509.MarshalECPrivateKey` and `x509.ParseECPrivateKey` which directly return an `*ecdsa.PrivateKey` btw 😇